src/topology/editor.rs

```diff
@@ -339,41 +402,167 @@ impl<'topology> TopologyEditor<'topology> {
         //         - By construction, flags are trusted to be in sync with the
         //           cpuset and nodeset params + only one of them is set as
         //           requested by hwloc
-        errors::call_hwloc_int_normal("hwloc_topology_allow", || unsafe {
+        let result = errors::call_hwloc_int_normal("hwloc_topology_allow", || unsafe {
             hwlocality_sys::hwloc_topology_allow(self.topology_mut_ptr(), cpuset, nodeset, flags)
-        })
-        .map(std::mem::drop)
-        .map_err(HybridError::Hwloc)
+        });
+        match result {
+            Ok(_) => Ok(()),
+            Err(RawHwlocError {
+                errno: Some(Errno(ENOSYS)),
+                ..
+            }) => Err(AllowSetError::Unsupported.into()),
+            Err(other) => Err(HybridError::Hwloc(other)),
+        }
     }
 
-    /// Add more structure to the topology by adding an intermediate [`Group`]
+    /// Add more structure to the topology by creating an intermediate [`Group`]
```

Comment:
Testing group creation led me to completely re-think my understanding of how groups work. Accordingly the docs, API and code were deeply rewritten.